src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/logging/AppInsights.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.logging;
+
+import com.google.common.collect.ImmutableMap;
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.azure.servicebus.IMessage;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.logging.appinsights.AbstractAppInsights;
+
+@Component
+public class AppInsights extends AbstractAppInsights {
+
+    static final String DEAD_LETTER_EVENT = "DeadLetter";
+
+    public AppInsights(TelemetryClient client) {
+        super(client);
+    }
+
+    public void trackDeadLetteredMessage(IMessage message, String queue, String reason, String description) {
+        telemetry.trackEvent(
+            DEAD_LETTER_EVENT,
+            ImmutableMap.of(
+                "reason", reason,
+                "description", description,
+                "messageId", message.getMessageId(),
+                "queue", queue
+            ),
+            ImmutableMap.of(
+                "deliveryCount", (double) (message.getDeliveryCount() + 1)
```

Comment:
why +1?

Reply:
Library delivers messages from `0`. Counter is increased once lock is expired and message can be re-consumed

Reply:
Maybe it be named something else than `deliveryCount` if it's not message's actual delivery count?

Reply:
It is delivery count. We use same wording and value in other logs too. It is rationally correct: when we read the logs, when we will read the events